app/core/Encryptor/Encryptor.ts

```diff
@@ -0,0 +1,236 @@
+import { NativeModules } from 'react-native';
+import { hasProperty, isPlainObject, Json } from '@metamask/utils';
+import {
+  BYTE_COUNT,
+  SHA256_DIGEST_LENGTH,
+  ENCRYPTION_LIBRARY,
+  DEFAULT_DERIVATION_PARAMS,
+  OLD_ITERATIONS_NUMBER,
+} from './constants';
+import type {
+  EncryptionResult,
+  KeyDerivationOptions,
+  GenericEncryptor,
+} from './types';
+
+const Aes = NativeModules.Aes;
+const AesForked = NativeModules.AesForked;
+
+/**
+ * Checks if the provided object is a `KeyDerivationOptions`.
+ *
+ * @param derivationOptions - The object to check.
+ * @returns Whether or not the object is a `KeyDerivationOptions`.
+ */
+const isKeyDerivationOptions = (
+  derivationOptions: unknown,
+): derivationOptions is KeyDerivationOptions =>
+  isPlainObject(derivationOptions) &&
+  hasProperty(derivationOptions, 'algorithm') &&
+  hasProperty(derivationOptions, 'params');
+
+/**
+ * The Encryptor class provides methods for encrypting and
+ * decrypting data objects using AES encryption with native libraries.
+ * It supports generating a salt, deriving an encryption key from a
+ * password and salt, and performing the encryption and decryption processes.
+ */
+class Encryptor implements GenericEncryptor {
+  key: string | null = null;
+
+  /**
+   * Generates a base64-encoded salt string.
+   * @param byteCount - The number of bytes for the salt. Defaults to 32.
+   * @returns The base64-encoded salt string.
+   */
+  private generateSalt(byteCount = BYTE_COUNT) {
+    const view = new Uint8Array(byteCount);
+    // @ts-expect-error - globalThis is not recognized by TypeScript
+    global.crypto.getRandomValues(view);
```

Comment:
When I removed this line, I didn't get any errors 🤔

```suggestion
    const view = new Uint8Array(byteCount);
    global.crypto.getRandomValues(view);
```